Build-process.md

```diff
@@ -28,45 +29,56 @@ The build process is performed by [`scripts/build.py`](https://github.com/oppia/
 There are two primary build modes: dev mode and prod mode. When running in prod mode, you can also enable maintenance mode or deploy mode, or you can choose to only minify third-party libraries. This means the build mode hierarchy looks like this:
 
 * Dev mode
+* Minify only mode
 * Prod mode
   * Maintenance mode
   * Deploy mode
-  * Minify only third-party libraries
 
-Note that maintenance and deploy mode can be enabled simultaneously.
+Note that the maintenance and deploy modes can be enabled simultaneously, so you should really think of them as flags. If you're in prod mode, you can also enable maintenance mode dev mode, both, or neither.
 
-We use prod mode for most everything besides local development. For example, we use prod mode when deploying to production and testing servers, and we run most tests in prod mode. A notable exception is the lighthouse accessibility tests, which run in dev mode because they don't depend on prod mode, and dev mode compilation is faster. Note that we also sometimes use prod mode locally when debugging to run a version of the app that's closer to the deployed version.
+Dev mode compiles the fastest, but it produces an application that's least like what we run in production. On the other hand, prod mode takes longer to compile but produces an application that is  the closest to what we run in production. Minify only mode is in the middle.
```

Comment:
Extra space after "is".

Perhaps, throughout, hyphenate "minify-only" so that "minify" is not read as a verb.

Reply:
done (both)